pluginlib/pluginlib-extras.cmake

```diff
@@ -45,9 +45,4 @@ find_package(tinyxml2_vendor REQUIRED)
 find_package(TinyXML2 REQUIRED)
 list(APPEND pluginlib_LIBRARIES ${TinyXML2_LIBRARIES})
 
-add_library(tinyxml2_vendor INTERFACE)
-target_include_directories(tinyxml2_vendor INTERFACE
-  ${TinyXML2_INCLUDE_DIRS})
-target_link_libraries(tinyxml2_vendor INTERFACE
-  ${TinyXML2_LIBRARIES})
-list(APPEND pluginlib_TARGETS tinyxml2_vendor)
+list(APPEND pluginlib_TARGETS tinyxml2)
```

Comment:
Looking at the CMake module https://github.com/ros2/tinyxml2_vendor/blob/c89337439ffe9b5123ca310af0d746eb9cbfa110/cmake/Modules/FindTinyXML2.cmake I don't you can safely assume that the target `tinyxml2` exists in all the cases. Therefore I think your use case needs a different fix.

Reply:
Ah, ok I missed that part. I reworked my fix to simply decorate `tinyxml2_vendor` as `IMPORTED`, which also fixed the CMake error.